commitments/mode.go

```diff
@@ -4,10 +4,22 @@ import (
 	"fmt"
 )
 
+// EncodingType represents the serialization format for the certificate
+type EncodingType byte
+
+const (
+	// RLPEncoding represents RLP encoding (default)
+	RLPEncoding EncodingType = iota
+	// ABIVerifyV2CertEncoding represents ABI encoding
+	ABIVerifyV2CertEncoding
+)
+
 type CommitmentMeta struct {
 	Mode CommitmentMode
 	// version is shared for all modes and denotes version of the EigenDA certificate
 	Version EigenDACommitmentType
+	// encoding type for the certificate (defaults to RLPEncoding)
+	Encoding EncodingType
```

Comment:
I like this like this, we use the zero value as a sane default. Prefer this over the pointer to encode optionality you used elsewhere. Not sure why we have 2 different ways of representing this